### PR TITLE
Issue #101: Jetpack live site tests

### PR DESCRIPTION
**Pull request by aforcier**
Addresses #101, adding some live tests for Jetpack sites.

* [x] A WP.com account with a single Jetpack site connected
* [x] A WP.com account with multiple Jetpack sites connected
* [x] A WP.com account with a Jetpack multisites connected
* [x] A self-hosted site without Jetpack (check isJetpack etc statuses)
* [x] A self-hosted site with Jetpack installed but not connected
* [x] A self-hosted site with Jetpack installed and connected
  * [x] User is not signed into WP.com
  * [x] Connected account is the one currently logged into (should throw `DUPLICATE_SITE` error: #285)
  * [x] Connected account is not the one currently logged into
* [x] A WP.com account with a Jetpack site that the user has already added as a self-hosted site (the self-hosted Jetpack site should be 'upgraded' to a WP.com Jetpack site: #285)

Also added a test for adding multiple self-hosted sites, including adding the same site twice (we didn't have a specific test for this before).

### CODE REVIEW

**Comment by aforcier** [comment]
The test in b26f3d7 brings up an edge case we should be aware of: if a user is signed into WP.com, and then adds a site over XML-RPC that is Jetpack-connected _but to a different WP.com account_, it is added to the store but is flagged as `isJetpack() == true`.

We can tell sites like these because if the user is signed into WP.com, any site that matches `!isWPCom() && isJetpack()` should be a Jetpack self-hosted site linked to a different WP.com account. But we might have a surprise, for example, if we list Jetpack sites based on `mSiteStore.getJetpackSites()`.

This case is mentioned in https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/207 - I'm not sure it's critical enough to address right away, but something we should probably think about (and guard against in the meantime).

**Comment by aforcier** [comment]
Ready for review - cc @maxme.

File: example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestJetpack.java [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+        assertEquals(1, mSiteStore.getSelfHostedSitesCount());
+        assertEquals(0, mSiteStore.getJetpackSitesCount());
+
+        SiteModel site = mSiteStore.getSites().get(0);
+
+        assertFalse(site.isJetpack());
+        assertEquals(0, site.getSiteId());
+    }
+
+    public void testXMLRPCJetpackSiteFetch() throws InterruptedException {
```
Comment (maxme) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestJetpack.java]: Can we rename this one to `testXMLRPCJetpackConnectedSiteFetch`?

File: example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestJetpack.java [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+        // Wait for a network response / onChanged event
+        assertTrue(mCountDownLatch.await(TestUtils.DEFAULT_TIMEOUT_MS, TimeUnit.MILLISECONDS));
+
+        // Expect no DB changes
+        assertEquals(1, mSiteStore.getSitesCount());
+        assertTrue(mSiteStore.hasJetpackSite());
+        assertTrue(mSiteStore.hasWPComSite());
+        assertFalse(mSiteStore.hasSelfHostedSite());
+
+        signOutWPCom();
```
Comment (maxme) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestJetpack.java]: +1

**Comment by maxme** [comment]
Tests look good, great you tested the sign out in `testXMLRPCJetpackToWPComDuplicateSiteFetch`, I didn't think about it but yes the "upgraded" site should be removed when the user signs out

File: example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestJetpack.java (hunk 1 above)
Comment (aforcier) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestJetpack.java]: Done in 628c94f.

**Comment by aforcier** [comment]
Ready to go @maxme!

**Comment by maxme** [comment]
:shipit:
